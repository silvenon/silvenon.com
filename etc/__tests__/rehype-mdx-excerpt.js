const mdx = require('@mdx-js/mdx')
const dedent = require('dedent')
const excerpt = require('../rehype-mdx-excerpt')

describe('excerpt', () => {
  it('includes content above the marker', async () => {
    await expect(
      mdx(
        dedent`
          import Foo from './components/foo'

          # Hello world!

          Included in the excerpt.

          <IgnoreJSX />

          <!-- excerpt -->

          Not included in the excerpt.
        `,
        {
          filepath: 'pages/blog/post.mdx',
          rehypePlugins: [[excerpt, 'excerpt']],
        },
      ),
    ).resolves.toMatchSnapshot()
  })

  it('fails if the marker is missing', async () => {
    await expect(
      mdx(
        dedent`
          import Foo from './components/foo'

          # Hello world!

          Oops, forgot to add the marker!

          <Foo />
        `,
        {
          filepath: 'pages/blog/post.mdx',
          rehypePlugins: [[excerpt, 'excerpt']],
        },
      ),
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      `"Unable to find the excerpt marker in pages/blog/post.mdx"`,
    )
  })

  it('supports including first X nodes', async () => {
    await expect(
      mdx(
        dedent`
          import Foo from './components/foo'

          # Hello world!

          Included in the excerpt.

          <IgnoreJSX />

          Not included in the excerpt.
        `,
        {
          filepath: 'pages/blog/post.mdx',
          rehypePlugins: [[excerpt, 3]],
        },
      ),
    ).resolves.toMatchSnapshot()
  })
})
