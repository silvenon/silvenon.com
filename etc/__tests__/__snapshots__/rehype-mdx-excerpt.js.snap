// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`excerpt includes content above the marker 1`] = `
"/* @jsx mdx */
import Foo from './components/foo'
export const Excerpt = function MDXExcerpt() {
  return <>
    <h1>{\`Hello world!\`}</h1>
    <p>{\`Included in the excerpt.\`}</p>

  </>;
};
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};
const IgnoreJSX = makeShortcode(\\"IgnoreJSX\\");
const layoutProps = {
  Excerpt
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">

    <h1>{\`Hello world!\`}</h1>
    <p>{\`Included in the excerpt.\`}</p>
    <IgnoreJSX mdxType=\\"IgnoreJSX\\" />

    <p>{\`Not included in the excerpt.\`}</p>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;"
`;

exports[`excerpt supports including first X nodes 1`] = `
"/* @jsx mdx */
import Foo from './components/foo'
export const Excerpt = function MDXExcerpt() {
  return <>
    <h1>{\`Hello world!\`}</h1>
    <p>{\`Included in the excerpt.\`}</p>

  </>;
};
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};
const IgnoreJSX = makeShortcode(\\"IgnoreJSX\\");
const layoutProps = {
  Excerpt
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">

    <h1>{\`Hello world!\`}</h1>
    <p>{\`Included in the excerpt.\`}</p>
    <IgnoreJSX mdxType=\\"IgnoreJSX\\" />
    <p>{\`Not included in the excerpt.\`}</p>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;"
`;
